vault/templates/secret.yaml

```diff
@@ -1,24 +1,29 @@
 {{- if and (not .Values.tls.secretName) (not .Values.certManager.certificate.enabled) }}
-{{- $cn := include "vault.fullname" . -}}
+{{- $cn := include "vault.fullname" $ -}}
 {{- $externalDNS := print $cn "." .Release.Namespace -}}
 {{- $ca := genCA "vault-ca" 365 -}}
 {{- $cert := genSignedCert $cn (list "127.0.0.1") (list $cn $externalDNS) 365 $ca -}}
+{{- $caNamespaces := list .Release.Namespace -}}
+{{- if .Values.tls.caNamespaces }}{{ $caNamespaces = (concat $caNamespaces .Values.tls.caNamespaces ) | uniq }}{{ end -}}
```

Comment:
`caNamespaces` can be declared in one line with both the default value + the specified list

Reply:
Do you perhaps mean the below?

```
{{- $caNamespaces := list .Release.Namespace -}}{{- if .Values.tls.caNamespaces }}{{ $caNamespaces = (concat $caNamespaces .Values.tls.caNamespaces ) | uniq }}{{ end -}}
```

Reply:
Or this?
```
{{- $caNamespaces := if .Values.tls.caNamespaces }}{{ (append (list .Release.Namespace) .Values.tls.caNamespaces) | uniq }}{{ else }}{{ (list .Release.Namespace) }}{{ end -}}
```

Reply:
Either way, I would still prefer the existing version, since using a single line for complex logic like this can make the code harder to read and maintain. It's generally recommended to use multiple lines for better readability and clarity.

Reply:
I mean something straightforward, not sure if this is an allowed syntax however.
`{{ $caNamespaces := (concat (list .Release.Namespace) .Values.tls.caNamespaces ) | uniq }}`.
We can keep as is, it doesn't differ in any way, and it's standardised.

Reply:
Hi @ramizpolic, that was also my approach first but if the caNamespaces object is empty or non-existent in values.yaml, it would throw an error that you try to concat a list with a null value (at least that was what I experienced if I remember correctly), hence the check...

Reply:
That's a nice catch, @akijakya. One would assume that concating a nil slice to an existing slice would work (thinking of golang append), but seems like it doesn't.